src/ducks/Signals/signalModal/SignalMasterModalForm.js

```diff
@@ -1,46 +1,82 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { connect } from 'react-redux'
-import { Modal, Button, Icon } from '@santiment-network/ui'
+import { Button, Icon } from '@santiment-network/ui'
+import Dialog from '@santiment-network/ui/Dialog'
 import SignalMaster from '../signalFormManager/SignalMaster'
 import { checkIsLoggedIn } from '../../../pages/UserSelectors'
 import styles from './SignalMasterModalForm.module.scss'
+import { push } from 'react-router-redux'
+import { SIGNAL_ROUTES } from '../common/constants'
 
 const SignalMasterModalForm = ({
   label = 'New signal',
   metaFormSettings,
   canRedirect = true,
-  isLoggedIn
-}) => (
-  <Modal
-    trigger={
-      <Button
-        variant='fill'
-        accent='positive'
-        disabled={!isLoggedIn}
-        className={styles.newSignal}
-      >
-        <Icon type='plus-round' className={styles.newSignal__icon} />
-        {label}
-      </Button>
+  redirectEnabled = false,
+  isLoggedIn,
+  redirect,
+  match
+}) => {
+  const [dialogOpenState, setDialogOpenState] = useState(redirectEnabled)
+  const [dialogTitle, onSetDialogTitle] = useState('')
+
+  const onClose = () => {
+    if (redirectEnabled) {
+      redirect()
+    } else {
+      setDialogOpenState(false)
     }
-    showDefaultActions={false}
-    title='Create signal'
-    classes={{ modal: styles.modalCentered }}
-  >
-    {closeModal => (
-      <SignalMaster
-        onClose={closeModal}
-        canRedirect={canRedirect}
-        metaFormSettings={metaFormSettings}
-      />
-    )}
-  </Modal>
-)
+  }
+
+  return (
+    <Dialog
+      open={dialogOpenState}
+      onClose={onClose}
+      onOpen={() => setDialogOpenState(true)}
+      trigger={signalModalTrigger(isLoggedIn, label, !redirectEnabled)}
+      title={dialogTitle}
+    >
+      <Dialog.ScrollContent className={styles.TriggerPanel}>
+        <SignalMaster
+          triggerId={match ? match.params.id : undefined}
+          setTitle={onSetDialogTitle}
+          onClose={() => setDialogOpenState(false)}
+          canRedirect={canRedirect}
+          metaFormSettings={metaFormSettings}
+        />
+      </Dialog.ScrollContent>
+    </Dialog>
+  )
+}
 
 const mapStateToProps = state => {
   return {
     isLoggedIn: checkIsLoggedIn(state)
   }
 }
 
-export default connect(mapStateToProps)(SignalMasterModalForm)
+const mapDispatchToProps = dispatch => ({
+  redirect: () => {
+    dispatch(push(SIGNAL_ROUTES.MY_SIGNALS))
+  }
+})
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(SignalMasterModalForm)
+
+const signalModalTrigger = (isLoggedIn, label, canShow) =>
+  !canShow ? (
+    <div />
```

Comment:
Just return `null`/`undefined`

Reply:
It is a trigger for dialog. Dialog crashes if trigger return null/undefined. Need to show dialog without a button/trigger for dialog.
![image](https://user-images.githubusercontent.com/14061779/59742068-6ed82a00-9275-11e9-996a-ba5c0b979b91.png)

Reply:
Then don't render `Dialog` itself in such case